src/tools/rustfmt/Cargo.toml

```diff
@@ -50,7 +50,7 @@ serde_json = "1.0"
 term = "0.7"
 thiserror = "1.0.40"
 toml = "0.7.4"
-tracing = { version = "0.1.37", default-features = false, features = ["std"] }
+tracing = { workspace = true }
```

Comment:
I don't think this is a valid change for rustfmt. Although rustfmt is part of the r-l/rust workspace, it's independently developed in https://github.com/rust-lang/rustfmt, and I'm pretty sure this would prevent us from build from source when working on the git subtree, which isn't structured as a cargo workspace. I'd like to opt out of all rustfmt changes from this PR, but feel free to open a new PR directly to https://github.com/rust-lang/rustfmt

Reply:
Ah, will update 👍

Reply:
@ytmimi is it sufficient to revert the Cargo.toml changes, or do the other changes need to go via the other repo as well?

Reply:
Also, there are some [other workspace members that are also subtrees](https://rustc-dev-guide.rust-lang.org/external-repos.html?#external-dependencies-subtrees), like clippy, and miri. I didn't look at any tools besides rustfmt, but it's possible maintainers of those tools would prefer PRs be opened up directly to their respective repos as well.

Reply:
@ytmimi happy to split if needed but the wording you link says:

> The exception is that when rustc changes are required to implement a new tool feature or test, that should happen in one collective rustc PR.

Reply:
> @ytmimi is it sufficient to revert the Cargo.toml changes, or do the other changes need to go via the other repo as well?

 I'd rather that all changes were reverted, thanks!

Reply:
Using workspace dependencies for the Cargo.toml files in `compiler` (except in codegen_gcc and codegen_clf is entirely fine).

The root problem here is not the use of workspace dependencies, the problem is that rustfmt has tracing in its Cargo.toml, which it shouldn't. It should use rustc's tracing instead.